File: 06/CSharp/TripleExpressionApp/TripleExpressionApp/Divide.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace TripleExpressionApp
{
    public class Divide : BinaryOperator
    {
        public Divide(ITripleExpression first, ITripleExpression second)
        {
            fOp = first;
            sOp = second;
        }

        override protected double Apply(double a, double b)
        {
            return a / b;
        }
    }
}

File: 05/Java/TripleExpression/Divide.java
package TripleExpression;

public class Divide extends BinaryOperator implements TripleExpression {
    public Divide(TripleExpression first, TripleExpression second) {
        super(first, second);
    }

    protected double apply(double a, double b) {
        return a / b;
    }
}